src/main/scala/com/fulcrumgenomics/phasing/HapCutToVcf.scala

```diff
@@ -0,0 +1,413 @@
+      likelihoods = tokens(1).split(",").map(_.toFloat).toList,
+      delta       = tokens(2).toFloat,
+      rMEC        = tokens(3).toFloat
+    )
+  }
+
+  /** The VCF header FORMAT keys that will be added for HapCut specific-genotype information. */
+  val formatHeaderKeys: Seq[String] = {
+    Seq(ReadCountFormatTag, LikelihoodChangeFormatTag, MaxLikelihoodChangeFormatTag, MecReductionFormatTag, PhaseSetFormatTag)
+  }
+
+  /** The VCF header FORMAT lines that will be added for HapCut specific-genotype information. */
+  val formatHeaderLines: Seq[VCFHeaderLine] = {
+    Seq(ReadCountFormatHeaderLine, LikelihoodChangeFormatHeaderLine, MaxLikelihoodChangeFormatHeaderLine, MecReductionFormatHeaderLine, PhaseSetFormatHeaderLine)
+  }
+
+  /** Adds the HapCut-specific genotype information to the given genotype builder. */
+  def addTo(builder: GenotypeBuilder, info: HapCutCallGenotypeInfo): GenotypeBuilder = {
+    builder
+      .attribute(ReadCountFormatTag, info.readCounts.asJava)
+      .attribute(LikelihoodChangeFormatTag, info.likelihoods.asJava)
+      .attribute(MaxLikelihoodChangeFormatTag, info.delta)
+      .attribute(MecReductionFormatTag, info.rMEC)
+  }
+}
+
+/** Information about a variant phased by HapCut.
+  *
+  * @param offset the 1-based offset of the variant relative to the variants given to HapCut.
+  * @param haplotype1 the first allele.
+  * @param haplotype2 the second allele.
+  * @param contig the contig.
+  * @param pos the position.
+  * @param ref the reference allele.
+  * @param alts the alternate alleles.
+  * @param genotype the genotype.
+  * @param info the HapCut-specific genotype information.
+  * @param firstInBlock true if this variant is the first variant in a phased block, false otherwise.
+  */
+private class HapCutCall private(val offset: Int,
+                                 val haplotype1: Int, val haplotype2: Int,
+                                 val contig: String, val pos: Int,
+                                 val ref: String, val alts: String,
+                                 val genotype: String, val info: HapCutCallGenotypeInfo,
+                                 val firstInBlock: Boolean
+                                )
+
+private object HapCutCall {
+
+  /** Parse a variant line. */
+  def apply(callLine: String, blockOffset: Int): HapCutCall = {
+    val tokens = callLine.split("\t")
+    val offset = tokens(0).toInt
+    new HapCutCall(
+      offset       = offset,
+      haplotype1   = tokens(1).toInt,
+      haplotype2   = tokens(2).toInt,
+      contig       = tokens(3),
+      pos          = tokens(4).toInt,
+      ref          = tokens(5),
+      alts         = tokens(6),
+      genotype     = tokens(7),
+      info         = HapCutCallGenotypeInfo(tokens(8)),
+      firstInBlock = blockOffset == offset
+    )
+  }
+
+  /** Converts the HapCut variant representation to a VariantContext. */
+  def toVariantContext(call: HapCutCall, sampleName: String): VariantContext = {
```

Comment:
I'm going to suggest that the following is an anti-pattern: methods in a companion object that take in, as a parameter, a single instance of the companion class and operate on it.